code/datums/repositories/decls.dm

```diff
@@ -40,6 +40,19 @@ var/global/repository/decls/decls_repository = new
 	RETURN_TYPE(/decl)
 	. = get_decl(fetched_decl_ids[decl_id], validate_decl_type)
 
+/repository/decls/proc/get_decl_by_id_or_var(var/decl_id, var/decl_prototype, var/check_var = "name")
```

Comment:
Given the code this is replacing, I feel like this should have a comment saying it's for grandfathered in items and shouldn't be used for new code without good reason.